neuralmonkey/learning_utils.py

```diff
@@ -122,100 +138,131 @@ def training_loop(tf_manager: TensorFlowManager,
+                                               epochs,
+                                               trainer_result,
                                                train=True)
                 else:
-                    tf_manager.execute(batch_dataset, [trainer],
-                                       train=True, summaries=False)
-
-                if step % validation_period == validation_period - 1:
-                    val_results, val_outputs = run_on_dataset(
-                        tf_manager, runners, val_dataset,
-                        postprocess, write_out=False,
-                        batch_size=runners_batch_size)
-                    # ensure val outputs are iterable more than once
-                    val_outputs = {k: list(v) for k, v in val_outputs.items()}
-                    val_evaluation = evaluation(
-                        evaluators, val_dataset, runners, val_results,
-                        val_outputs)
-
-                    this_score = val_evaluation[main_metric]
-                    tf_manager.validation_hook(this_score, epoch_n, batch_n)
-
-                    log("Validation (epoch {}, batch number {}):"
-                        .format(epoch_n, batch_n), color='blue')
-
-                    _log_continuous_evaluation(tb_writer, tf_manager,
-                                               main_metric,
-                                               val_evaluation,
-                                               seen_instances, epoch_n,
-                                               epochs,
-                                               val_results, train=False)
-
-                    if this_score == tf_manager.best_score:
-                        best_score_str = colored(
-                            "{:.4g}".format(tf_manager.best_score),
-                            attrs=['bold'])
-                    else:
-                        best_score_str = "{:.4g}".format(tf_manager.best_score)
-
-                    log("best {} on validation: {} (in epoch {}, "
-                        "after batch number {})"
-                        .format(main_metric, best_score_str,
-                                tf_manager.best_score_epoch,
-                                tf_manager.best_score_batch),
-                        color='blue')
-
-                    log_print("")
-                    _print_examples(val_dataset, val_outputs,
-                                    val_preview_input_series,
-                                    val_preview_output_series,
-                                    val_preview_num_examples)
+                    tf_manager.execute(batch_dataset,
+                                       [trainer],
+                                       train=True,
+                                       summaries=False)
+
+                for validation_dataset in val_datasets:
+                    if step % validation_period == validation_period - 1:
```

Comment:
jj toho jsem si uz vsiml, ale jelikoz jak jsme resili vyse s jindrichem se tahle vetev bude zahazovat tak uz jsem to nekomitoval